src/estimator/diffusion.rs

```diff
@@ -54,20 +51,34 @@ impl<U: EosUnit, E: EquationOfState + EntropyScaling<U>> DataSet<U, E> for Diffu
         vec!["temperature", "pressure"]
     }
 
-    fn predict(&self, eos: &Rc<E>) -> Result<QuantityArray1<U>, EstimatorError>
+    fn predict(&self, eos: &Arc<E>) -> Result<QuantityArray1<U>, EstimatorError>
     where
         QuantityScalar<U>: std::fmt::Display + std::fmt::LowerExp,
     {
-        let unit = self.target.get(0);
-        let mut prediction = Array1::zeros(self.datapoints) * unit;
         let moles = arr1(&[1.0]) * U::reference_moles();
-        for i in 0..self.datapoints {
-            let t = self.temperature.get(i);
-            let p = self.pressure.get(i);
-            let state = State::new_npt(eos, t, p, &moles, DensityInitialization::None)?;
-            prediction.try_set(i, state.diffusion()?)?;
-        }
-        Ok(prediction)
+        let ts = self
+            .temperature
+            .to_reduced(U::reference_temperature())
```

Comment:
As this is not parallel at the moment, there is no need to use dimensionless variables

Reply:
True. This change will be needed when we introduce parallel evaluation of the model prediction (and want to reduce code repetition) because iteration over `SIArray` objects is currently not flexible enough. 

I can change it back to how it was, if you have strong feelings about this implementation for the time being.